spec/models/student_spec.rb

```diff
@@ -321,6 +321,34 @@
       end
     end
 
+    context 'where there is a grace deduction' do
+      before do
+        # setting up an assignment to use grace credits on
+        @assignment = create(:assignment)
+        @grouping = create(:grouping, assignment: @assignment)
+        @student1 = create(:student)
+        @student2 = create(:student)
+        @membership1 = create(:student_membership, role: @student1, grouping: @grouping,
+                                                   membership_status: StudentMembership::STATUSES[:accepted])
+        @membership2 = create(:student_membership, role: @student2, grouping: @grouping,
+                                                   membership_status: StudentMembership::STATUSES[:inviter])
+        create(:grace_period_deduction, membership: @membership1, deduction: 2)
+      end
+
+      it 'returns the correct value of used credits per assessment' do
+        expect(@student1.grace_credits_used_for(@assignment)).to eq(2)
+      end
+
+      it 'updates remaining grace credits after deductions' do
+        expect(@student1.remaining_grace_credits).to eq(3)
+      end
+
+      # it 'deducts grace credits from each group member' do  # How to make this testcase..?
```

Comment:
I'm attempting to create a grace credit deduction for a group assignment, and then check that each member in the group has received the deduction. I simulated this on my own, and it works as expected, but I want to be able to explicitly test for this with RSpec.
I'm unable to recreate this scenario within RSpec, as the grace credit deductions are applied to each membership, rather than a group. There may be another method to apply grace credit deductions on a group basis, but I have not found that method. I tried to investigate how the grace credit deductions are applied to a group, but I'm not able to trace it through fully.
I found that the `SubmissionsJob` method `perform` in `app/jobs/submissions_job.rb` runs when an instructor collects assignments and applies the grace deduction. I was hoping to find some sort of for-loop that applies the grace credit deduction equally to each member within a grouping so that I could mimic that behaviour within my test case.